cargo-afl/build.rs

```diff
@@ -14,67 +13,90 @@ static AR_CMD: &str = "ar";
+    assert!(status.success());
+
+    let work_dir = PathBuf::from(AFL_SRC_PATH);
 
-    let base = if building_on_docs_rs {
-        Some(out_dir)
-    } else {
-        None
-    };
+    let base: Option<&Path> = None;
 
     // smoelius: Lock `work_dir` until the build script exits.
     #[cfg(unix)]
     let _file = sys::lock_path(&work_dir).unwrap();
 
-    build_afl(&work_dir, base.as_deref());
-    build_afl_llvm_runtime(&work_dir, base.as_deref());
+    let mut llvm_config = String::new();
+
+    if cfg!(feature = "cmplog") {
+        llvm_config = common::get_llvm_config();
+    }
+
+    build_afl(&work_dir, base, llvm_config);
+
+    if cfg!(feature = "cmplog") {
+        build_afl_llvm_runtime(&work_dir, base);
+    }
 }
 
-fn build_afl(work_dir: &Path, base: Option<&Path>) {
+fn build_afl(work_dir: &Path, base: Option<&Path>, llvm_config: String) {
+    if cfg!(feature = "cmplog") {
+        // Make sure we are on nightly for the -Z flags
+        assert!(
+            rustc_version::version_meta().unwrap().channel == rustc_version::Channel::Nightly,
+            "cargo-afl must be compiled with nightly for the cmplog feature"
+        );
+
+        // check if llvm tools are installed and with the good version for the plugin compilation
+        let mut command = Command::new(llvm_config.clone());
+        command.args(["--version"]);
+        let status = command
+            .status()
+            .unwrap_or_else(|_| panic!("could not run {llvm_config} --version"));
+        assert!(status.success());
+    }
+
+    // if you had already installed cargo-afl previously you **must** clean AFL++
     let mut command = Command::new("make");
     command
         .current_dir(work_dir)
-        .args(["clean", "all", "install"])
+        .args(["clean"])
         // skip the checks for the legacy x86 afl-gcc compiler
         .env("AFL_NO_X86", "1")
         // build just the runtime to avoid troubles with Xcode clang on macOS
-        .env("NO_BUILD", "1")
+        //.env("NO_BUILD", "1")
         .env("DESTDIR", common::afl_dir(base))
         .env("PREFIX", "")
+        .env("LLVM_CONFIG", llvm_config.clone())
         .env_remove("DEBUG");
-    let status = command.status().expect("could not run 'make'");
+
+    let status = command.status().expect("could not run 'make clean'");
+    assert!(status.success());
+
+    let mut command = Command::new("make");
+    command
+        .current_dir(work_dir)
+        .args(["install"])
+        // skip the checks for the legacy x86 afl-gcc compiler
+        .env("AFL_NO_X86", "1")
+        // build just the runtime to avoid troubles with Xcode clang on macOS
+        //.env("NO_BUILD", "1")
+        .env("DESTDIR", common::afl_dir(base))
+        .env("PREFIX", "")
+        .env("LLVM_CONFIG", llvm_config)
+        .env_remove("DEBUG");
+    let status = command.status().expect("could not run 'make install'");
     assert!(status.success());
```

Comment:
I think I talked with @vanhauser-thc about this and those options were unnecessary and we removed them. The added the clean later on, which was also unnecessary. Actually if you `cargo install --git https://github.com/rust-fuzz/afl.rs` it will not work, as cargo doesn't handle redirection properly in the submodules of AFL++. I tried working around it with doing the submodule init in the buildscript.

Reply:
> `cargo install --git https://github.com/rust-fuzz/afl.rs`

Is that why the PR involves submodules? To enable above command?

If so, is enabling the above command needed to enable the `cmplog` feature? (It feels to me like it's orthogonal.)

Reply:
Not exactly, I tried using `cargo install --git https://github.com/rust-fuzz/afl.rs` which didn't work as the submodule didn't redirect properly. Tried hotpatching this in the buildscript, which was a bad idea and forgot to remove it. I removed it in the last commit

Reply:
Recombined them, but removed the unnecessary "all"